File: 백준/Silver/14911. 궁합 쌍 찾기/궁합 쌍 찾기.py
numbers = list(map(int, input().split()))
target = int(input())

pairs = list()

for i in range(len(numbers)):
    for j in range(i + 1, len(numbers)):
        if i != j and numbers[i] +numbers[j] == target:
            pairs.append((min(numbers[i], numbers[j]), max(numbers[i], numbers[j])))

pairs = list(set(pairs))
pairs.sort()

for pair in pairs:
    print(*pair)
print(len(pairs))

File: 백준/Silver/14911. 궁합 쌍 찾기/궁합 쌍 찾기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        String[] numbersStr = sc.nextLine().split(" ");
        int[] numbers = new int[numbersStr.length];
        for (int i = 0; i < numbersStr.length; i++) {
            numbers[i] = Integer.parseInt(numbersStr[i]);
        }

        int target = sc.nextInt();
        sc.close();

        Set<String> pairsSet = new HashSet<>();

        for (int i = 0; i < numbers.length; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                if (numbers[i] + numbers[j] == target) {
                    int min = Math.min(numbers[i], numbers[j]);
                    int max = Math.max(numbers[i], numbers[j]);
                    pairsSet.add(min + " " + max);
                }
            }
        }

        List<String> pairsList = new ArrayList<>(pairsSet);
        Collections.sort(pairsList);

        for (String pair : pairsList) {
            System.out.println(pair);
        }

        System.out.println(pairsList.size());
    }
}